src/route.js

```diff
@@ -28,11 +28,13 @@ export default MnObject.extend(
       if (this.view && this.updateView(transition)) return
       let ViewClass = this.viewClass || View
       let viewOptions = _.result(this, 'viewOptions', {})
-      if (!(ViewClass.prototype instanceof View)) {
+      if (!(ViewClass.prototype instanceof View) &&
+        !(ViewClass.prototype instanceof CollectionView)) {
```

Comment:
```suggestion
      if (!(ViewClass.prototype instanceof View) &&
        !(ViewClass.prototype instanceof CollectionView)) {
```

add a isMarionetteView function and use it here and below

const isMarionetteView = proto => proto instanceof View || proto instanceof CollectionView

Reply:
Added a commit with that defined inline - if you want something like `Route._viewClassIsMarionetteView`, I can also do that.